### PR TITLE
YLP-367 Add fetch TIR

### PR DESCRIPTION
**Pull request by yvon-dblg**
- Use tide-whisperer API v0 as a fallback (much slower)
- Fetch TIR informations only when a patient is displayed

### CODE REVIEW

File: packages/yourloops/lib/data/api.ts [hunk 1]
```diff
@@ -33,7 +33,7 @@ import _ from "lodash";
 import { PatientData } from "models/device-data";
 import { MessageNote, MessagesThread } from "models/message";
 import { HttpHeaderKeys, HttpHeaderValues } from "../../models/api";
-import { User, UserRoles } from "../../models/shoreline";
+import { User, UserRoles, ComputedTIR } from "../../models/shoreline";
```
Comment (Nas84) [inline packages/yourloops/lib/data/api.ts:36]: the computedTIR should be part of shoreline. I think we should create a folder medicaldata (or maybe just data) cointining these declarations.

File: packages/yourloops/lib/team/hook.tsx [hunk 2]
```diff
@@ -29,9 +29,10 @@
 import * as React from "react";
 import bows from "bows";
 
+import { MedicalData } from "../../models/shoreline";
```
Comment (Nas84) [inline packages/yourloops/lib/team/hook.tsx:32]: Same comment as abose, the medical data is not part of shoreline

File: packages/yourloops/lib/team/hook.tsx [hunk 3]
```diff
@@ -29,9 +29,10 @@
...
 import { ITeam, TeamType, TeamMemberRole, TypeTeamMemberRole, TeamMemberStatus, ITeamMember } from "../../models/team";
 
-import { errorTextFromException } from "../utils";
+import { errorTextFromException, isUserAPatient } from "../utils";
```
Comment (Nas84) [inline packages/yourloops/lib/team/hook.tsx:35]: Utils file start to contains too much different responsibilities. What do you think if we start to split it ? i could start it if you would like.

File: packages/yourloops/lib/team/models.ts [hunk 4]
```diff
@@ -27,7 +27,7 @@
  */
 
 import { PostalAddress } from "../../models/generic";
-import { User } from "../../models/shoreline";
+import { User, MedicalData } from "../../models/shoreline";
```
Comment (Nas84) [inline packages/yourloops/lib/team/models.ts:30]: Same comment as above

File: packages/yourloops/models/shoreline.ts [hunk 5]
```diff
@@ -73,6 +74,42 @@ interface Preferences {
   displayLanguageCode?: "en" | "de" | "es" | "fr" | "it" | "nl";
   patientsStarred?: string[];
 }
+
+interface ComputedTIR {
```
Comment (Nas84) [inline packages/yourloops/models/shoreline.ts:78]: Same comment as above

**Review by Nas84 — CHANGES_REQUESTED** [review]

File: packages/yourloops/lib/data/api.ts (hunk 1 above)
Comment (yvon-dblg) [reply packages/yourloops/lib/data/api.ts:36]: shouldn't ?

File: packages/yourloops/lib/team/hook.tsx (hunk 3 above)
Comment (yvon-dblg) [reply packages/yourloops/lib/team/hook.tsx:35]: This function is removed in a following PR.
But yes this file has many topics, but for now it's still very small, without really a specific domain.
I think we can wait to split this file.

**Review by Nas84 — APPROVED** [review]

**Pull request merged**